cwltool/job.py

```diff
@@ -414,11 +414,17 @@ def get_tree_mem_usage(memory_usage):
 
 class CommandLineJob(JobBase):
     def run(self,
-            runtimeContext  # type: RuntimeContext
+            runtimeContext,         # type: RuntimeContext
+            TMPDIR_LOCK=None        # type: threading.Lock
             ):  # type: (...) -> None
 
-        if not os.path.exists(self.tmpdir):
-            os.makedirs(self.tmpdir)
+        if TMPDIR_LOCK != None:
+            if not os.path.exists(self.tmpdir):
```

Comment:
isn't this missing `with tmpdirLock:` before it?

Reply:
Yes, my mistake, I had included it in the version I was testing with but didn't commit it here.